tests/test_topology.py

```diff
@@ -25,22 +25,15 @@ def test_relative_free_energy_forcefield():
     bt0 = topology.BaseTopology(mol, ff0)
     bt1 = topology.BaseTopology(mol, ff1)
 
-    ordered_handles = ff0.get_ordered_handles()
-    bond_idx = ordered_handles.index(ff0.hb_handle)
-    angle_idx = ordered_handles.index(ff0.ha_handle)
-    proper_idx = ordered_handles.index(ff0.pt_handle)
-    improper_idx = ordered_handles.index(ff0.it_handle)
-    charge_idx = ordered_handles.index(ff0.q_handle)
-    lj_idx = ordered_handles.index(ff0.lj_handle)
-    ff0_params = ff0.get_ordered_params()
-    ff1_params = ff1.get_ordered_params()
-
-    combined_params = combine_ordered_params(ff0, ff1)
+    ff0_params = ff0.get_params()
+    ff1_params = ff1.get_params()
+
+    combined_params = combine_params(ff0_params, ff1_params)
     combined_qlj_params, combined_ubp = fftop.parameterize_nonbonded(
-        combined_params[charge_idx], combined_params[lj_idx]
+        combined_params.q_params, combined_params.lj_params
```

Comment:
Naming question: `ff_params.q_params` seems a bit redundant, but `ff_params.q` is a bit weird too (though maybe better?)

Reply:
> this can contain either "generalizable" forcefield params (of length # types) or "system params" (of length # atoms or # interactions)

I didn't (intentionally) change the representation of system params in this PR; `ForcefieldParams` was intended to only be used for the former kind. The addition of the new dataclass instead of adding to `Forcefield` was to have a type that is 1-1 with the existing representation as a list of parameter arrays. `ForcefieldParams` is generic because sometimes (e.g. in `RelativeFreeEnergyForcefield`) the parameters are tuples of (src, dst).